ui/app/src/components/RenameDashboardDialog/RenameDashboardDialog.tsx

```diff
@@ -0,0 +1,111 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ChangeEvent, Dispatch, DispatchWithoutAction, useState } from 'react';
+import { Dialog, DialogTitle, DialogContent, DialogActions, Button, TextField, IconButton } from '@mui/material';
+import { DashboardResource } from '@perses-dev/core';
+import CloseIcon from 'mdi-material-ui/Close';
+import { dashboardDisplayName } from '@perses-dev/core/dist/utils/text';
+
+export interface RenameDashboardDialogProps {
+  open: boolean;
+  onClose: DispatchWithoutAction;
+  onConfirm: Dispatch<string>;
+  dashboard: DashboardResource;
+}
+
+/**
+ * Dialog used to rename a dashboard.
+ * @param props.open Define if the dialog should be opened or not.
+ * @param props.closeDialog Provides the function to close itself.
+ * @param props.onConfirm Action to perform when user confirmed.
+ * @param props.dashboard The dashboard resource to rename.
+ * @constructor
+ */
+const RenameDashboardDialog = (props: RenameDashboardDialogProps) => {
+  const { open, onClose, onConfirm, dashboard } = props;
+  const [name, setName] = useState<string>();
+  const [error, setError] = useState<string>();
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
```

Comment:
I'm not sure we want to copy [how Grafana does it](https://github.com/grafana/grafana/issues/7883), but when renaming a dashboard, they append a second display name with the new dashboard name in the URL. 

Maybe there are other solutions to consider too like redirecting the old URL to avoid the path getting out of sync with the dashboard name (not important to address here, retaining the original URL like it's currently is definitely safest for now):
![image](https://user-images.githubusercontent.com/9369625/215147247-9a33c03d-a8c1-4e2a-a65c-f5c60c938f06.png)

Reply:
Spotted ! ^^
![image](https://user-images.githubusercontent.com/10258608/215815823-7441936f-ab68-485c-a3ed-0c25604fbfb8.png)

AFAIK, Perses is using the name as uid currently where grafana has a <uid>/<slug> in the URL + a display name. It's indeed an interesting topic, that would deserve its own discussion. 

Fun fact, you can still update the uid in grafana through the json "expert mode" let's say and if you do so, and validate you have a 404 error ^^